pilot/pkg/serviceregistry/kube/controller/controller_test.go

```diff
@@ -25,6 +25,7 @@ import (
 	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
 	coreV1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
+	discoveryv1alpha1 "k8s.io/api/discovery/v1alpha1"
```

Comment:
We should find a way to test both codepaths. I don't think that's happening now ... were you assuming separate test targets with/without the env variable set?

Reply:
@howardjohn any thoughts on this? Do you see any value in running these tests with both endpoint controllers?

Reply:
ack, yes we should run all of them with both settings. Was working on this next.

We also want to get some integration tests running but haven't set that up yet.

In a previous commit I turned EndpointSlice on by default and every test, including E2E, passed, so we do have some manual one time verification in the mean time